core/src/main/java/org/apache/iceberg/view/SQLViewRepresentationParser.java

```diff
@@ -0,0 +1,104 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.util.JsonUtil;
+
+class SQLViewRepresentationParser {
+  private static final String SQL = "sql";
+  private static final String DIALECT = "dialect";
+  private static final String SCHEMA_ID = "schema-id";
+  private static final String DEFAULT_CATALOG = "default-catalog";
+  private static final String DEFAULT_NAMESPACE = "default-namespace";
+  private static final String FIELD_ALIASES = "field-aliases";
+  private static final String FIELD_COMMENTS = "field-comments";
+
+  private SQLViewRepresentationParser() {}
+
+  static void toJson(SQLViewRepresentation view, JsonGenerator generator) throws IOException {
+    Preconditions.checkArgument(view != null, "Invalid view representation: null");
+    generator.writeStartObject();
+    generator.writeStringField(ViewRepresentationParser.TYPE, view.type());
+    generator.writeStringField(SQL, view.query());
+    generator.writeStringField(DIALECT, view.dialect());
+    generator.writeNumberField(SCHEMA_ID, view.schema().schemaId());
+
+    if (view.defaultCatalog() != null) {
+      generator.writeStringField(DEFAULT_CATALOG, view.defaultCatalog());
+    }
+
+    if (view.defaultNamespace() != null) {
+      JsonUtil.writeStringArray(
+          DEFAULT_NAMESPACE, Arrays.asList(view.defaultNamespace().levels()), generator);
+    }
+
+    if (view.fieldAliases() != null && !view.fieldAliases().isEmpty()) {
+      JsonUtil.writeStringArray(
+          SQLViewRepresentationParser.FIELD_ALIASES, view.fieldAliases(), generator);
+    }
+
+    if (view.fieldComments() != null && !view.fieldComments().isEmpty()) {
+      JsonUtil.writeStringArray(
+          SQLViewRepresentationParser.FIELD_COMMENTS, view.fieldComments(), generator);
+    }
+
+    generator.writeEndObject();
+  }
+
+  static SQLViewRepresentation fromJson(JsonNode node, List<Schema> schemas) {
+    int schemaId = JsonUtil.getInt(SCHEMA_ID, node);
```

Comment:
does this need a null check for `node`?

Reply:
If it was null it would have already been caught in ViewRepresentationParser, but we can add the check here as well so SQLViewRepresentationParser can stand alone properly.